### PR TITLE
Instrumentation.AWSLambda: Rename to remove .Contrib & minor cleanups.

### PR DESCRIPTION
**Pull request by Oberon00**
Required before the release #590.

## Changes
* BREAKING: Rename package (& namespaces) to remove .Contrib.
* BREAKING: Move public class AWSLambdaWrapper out of implementation subnamespace
* BREAKING (technically): Make AWSLambdaWrapper a static class (found through #607).
* Add version to ActivitySource

For significant contributions please make sure you have completed the following items:

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [X] Design discussion issue: #196 (see also <https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/590#issuecomment-1219051107>, CC @Kielek)

CC @srprash @rypdal

### CODE REVIEW

**Review by Kielek — APPROVED** [review]
Any chance to make same changes for other AWS packages?

**Comment by Oberon00** [comment]
@Kielek I'm only the owner of the AWS Lambda class, I think @srprash is responsible for AWS, and @srprash and @lupengamzn for X-Ray: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/58cfd202ffad0079e3d829c7ada023aa17e2187d/.github/component_owners.yml#L6-L10

**Review by utpilla — APPROVED** [review]

**Comment by utpilla** [comment]
Please update the `component_owners.yml` file as well.

**Comment by Oberon00** [comment]
@utpilla Done in 46464c2, I also noticed a few other references. I left the label name in the issue template as-is. I can change that as well if you like, but I think a maintainer should rename the label in the GitHub repository settings before.

File: .github/ISSUE_TEMPLATE/comp_contrib_instrumentation_awslambda.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ---
-name: OpenTelemetry.Contrib.Instrumentation.AWSLambda
-about: Issue with OpenTelemetry.Contrib.Instrumentation.AWSLambda
+name: OpenTelemetry.Instrumentation.AWSLambda
+about: Issue with OpenTelemetry.Instrumentation.AWSLambda
 labels: comp:contrib.instrumentation.awslambda
```
Comment (Oberon00) [inline .github/ISSUE_TEMPLATE/comp_contrib_instrumentation_awslambda.md:4]: See https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/593#issuecomment-1228237203:

If we want to change this to

```suggestion
labels: comp:instrumentation.awslambda
```

then a maintainer should rename this in the GitHub repository settings first.

**Comment by Oberon00** [comment]
Build failed due to unrelated problem which might be a race condition:

<detail>
```
[xUnit.net 00:00:02.74]     OpenTelemetry.Instrumentation.Wcf.Tests.TelemetryClientMessageInspectorTests.OutgoingRequestInstrumentationTest(instrument: True, filter: False, suppressDownstreamInstrumentation: False, includeVersion: False, enrich: False, enrichmentException: False, emptyOrNullAction: False) [FAIL]
  Failed OpenTelemetry.Instrumentation.Wcf.Tests.TelemetryClientMessageInspectorTests.OutgoingRequestInstrumentationTest(instrument: True, filter: False, suppressDownstreamInstrumentation: False, includeVersion: False, enrich: False, enrichmentException: False, emptyOrNullAction: False) [1 ms]
  Error Message:
   System.ObjectDisposedException : Cannot access a disposed object.
Object name: 'System.Net.HttpListener'.
  Stack Trace:
     at System.Net.HttpListener.CheckDisposed()
   at System.Net.HttpListener.Stop()
   at OpenTelemetry.Instrumentation.Wcf.Tests.TelemetryClientMessageInspectorTests..ctor() in D:\a\opentelemetry-dotnet-contrib\opentelemetry-dotnet-contrib\test\OpenTelemetry.Instrumentation.Wcf.Tests\TelemetryClientMessageInspectorTests.cs:line 56
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
```
</detail>

**Review by cijothomas — APPROVED** [review]

**Pull request merged**